model/src/main/scala/za/co/absa/atum/model/types/Measurement.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.atum.model.types
+
+import za.co.absa.atum.model.ResultValueType
+import za.co.absa.atum.model.dto.MeasurementDTO
+
+trait Measurement {
+  type T
+  def measureName: String
+  def measuredColumns: Seq[String]
+  def valueType: ResultValueType
+  def value: T
+  def stringValue: String
+}
+
+object Measurement {
+
+  def apply[T](from: MeasurementDTO): Measurement = {
```

Comment:
this has some similarity with agent's model/MeasureResult - did you check if it can be reused for Reader usages? Maybe slightly changed/moved if needed. But I think if model and agent are doing conceptually similar thing maybe we can unite and reuse some of the existing code

Reply:
The similarity did occur to me. But they are not 100% same unlike `AtumPartitions` and `AdditionalData` and did want to further delay in search for commonality. I think that can be done in future, if found useful.